src/libopensc/card-epass2003.c

```diff
@@ -397,16 +396,21 @@ aes128_encrypt_cmac_ft(struct sc_card *card, const unsigned char *key, int keysi
 	if(length < 16){
 		memcpy(&data2[0],input,length);
 		data2[length] = 0x80;
+	
+		//k2 xor padded data
+		for (int i=0;i<16;i++){
+			data2[i]=data2[i]^k2Bin[offset + i];
+		}
+	}else{
+		memcpy(&data2[0],input,length);
```

Comment:
you should make sure the length is less than 32 before calling memcpy, otherwise you overrun the `data2` buffer.